### PR TITLE
cargo-workspaces: init at 0.2.34

### PR DESCRIPTION
**Pull request by cjab**
###### Description of changes

Add [cargo-workspaces](https://github.com/pksunkara/cargo-workspaces).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/cargo-workspaces/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    sha256 = "sha256-7oJ/jjrv2XbOyLKUqp7EMFGR+czNSXCpWI8/JX5kmH8=";
+  };
+
+  cargoSha256 = "sha256-vZyodSxQd89dyaw4lr8vsbNjztXhFVgD0FcBP8KdPss=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
+
+  sourceRoot = "source/cargo-workspaces";
+  doCheck = false;
```
Comment (applePrincess) [inline pkgs/development/tools/rust/cargo-workspaces/default.nix:20]: Why skipping tests?

Comment (cjab) [reply pkgs/development/tools/rust/cargo-workspaces/default.nix]: Good question! After a bit of struggle tests are now running 😄.

File: pkgs/development/tools/rust/cargo-workspaces/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  cargoSha256 = "sha256-vZyodSxQd89dyaw4lr8vsbNjztXhFVgD0FcBP8KdPss=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
+
+  sourceRoot = "source";
+  cargoRoot = pname;
+  buildAndTestSubdir = pname;
```
Comment (cjab) [inline pkgs/development/tools/rust/cargo-workspaces/default.nix:26]: This repo is a little bit strange in that the Cargo.lock does not live at the root. Also, fixtures used by tests live in the root of the repo which caused some problems running tests. Setting `sourceRoot` and `cargoRoot` seemed to fix the issue for the patch phase but `buildAndTestSubdir` was also required for the build phase and check phases.

**Comment by cjab** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/188651

**Pull request closed without merging**